earlydisplay/src/main/java/net/neoforged/fml/earlydisplay/DisplayWindow.java

```diff
@@ -299,13 +298,10 @@ public String getGLVersion() {
         return this.glVersion;
     }
 
+    private final ReentrantLock crashLock = new ReentrantLock();
+
     private void crashElegantly(String errorDetails) {
-        String qrText;
-        try (var is = new BufferedReader(new InputStreamReader(getClass().getResourceAsStream("/glfailure.txt")))) {
-            qrText = is.lines().collect(Collectors.joining("\n"));
-        } catch (IOException ioe) {
-            qrText = "";
-        }
+        crashLock.lock(); // Crash at most once!
```

Comment:
A bit of a nitpick: Would using a `Semaphore` one permit be a bit more idiomatic? Taking a lock and not releasing it is a bit weird.